make_UPD_igv.py

```diff
@@ -0,0 +1,176 @@
+#! /usr/bin/env python3
+import sys
+import argparse
+import vcf
+
+def parse_ped(ped_file):
+    samples = {}  # 'sample_id': {'family': 'fam_id', 'parents': ['sample_id', 'sample_id']}
+    for line in ped_file:
+        ped_data = line.strip().split()
+        family, sample, father, mother, sex, phenotype = ped_data
+
+        # Create samples
+        if sample not in samples:
+            samples[sample] = {'family': family, 'parents': [], 'children': []}
+        if father != '0' and father not in samples:
+            samples[father] = {'family': family, 'parents': [], 'children': []}
+        if mother != '0' and mother not in samples:
+            samples[mother] = {'family': family, 'parents': [], 'children': []}
+
+        # Save sample relations
+        if father != '0':
+            samples[sample]['parents'].append(father)
+            samples[father]['children'].append(sample)
+        if mother != '0':
+            samples[sample]['parents'].append(mother)
+            samples[mother]['children'].append(sample)
+
+    families = {}
+    for sample in samples:
+        if len(samples[sample]['parents']) == 2:
+            families[sample] = samples[sample]['parents']
+
+    return samples, families
+
+
+def parse_vcf(vcf_file):  # returns list with genotypes
+    with open(vcf_file, 'r') as vcf_input_file:
+        vcf_reader = vcf.Reader(open(vcf_file, 'r'))
+        sampleid = vcf_reader.samples[0]
+        snv_list = []
+        for record in vcf_reader:
+            chrom = record.CHROM
+            pos = record.POS
+            dp = record.genotype(sampleid)['DP']
+            if dp >= args.mindepth and chrom is not "Y" :
+                if record.genotype(sampleid).phased: # sort genotype in unphased state.
+                    gt = record.genotype(sampleid)['GT'].split("|")
+                    gt.sort()
+                    gt = "/".join(gt)
+                else:
+                    gt = record.genotype(sampleid)['GT']
+                if len(record.ALT) > 1: # skip multiallelic position
+                   continue
+                allele = [str(record.REF[0]),str(record.ALT[0])]
+                variant_call = record.genotype(sampleid).is_variant
+                if variant_call is not None:
+                    genotype = [] 
+                    for item in gt.split("/"):
+                        genotype.append(allele[int(item)])
```

Comment:
you first split gt on |, and sort and joined them. and then split them again on / ?

Reply:
Yes, but in case gt does not contain phasing information ("|"), regular gt is used (which is joined with "/").
In either way:  one of the routes (phased or normal) either need an additional split of join. Code wise no difference.